addons/purchase/purchase.py

```diff
@@ -998,17 +998,17 @@ def _get_uom_id(self, cr, uid, context=None):
             return False
 
     _columns = {
-        'name': fields.text('Description', required=True),
-        'product_qty': fields.float('Quantity', digits_compute=dp.get_precision('Product Unit of Measure'), required=True),
-        'date_planned': fields.datetime('Scheduled Date', required=True, select=True),
-        'taxes_id': fields.many2many('account.tax', 'purchase_order_taxe', 'ord_id', 'tax_id', 'Taxes'),
-        'product_uom': fields.many2one('product.uom', 'Product Unit of Measure', required=True),
-        'product_id': fields.many2one('product.product', 'Product', domain=[('purchase_ok','=',True)], change_default=True),
+        'name': fields.text('Description', required=True, readonly=True, states={'draft': [('readonly', False)]}),
+        'product_qty': fields.float('Quantity', digits_compute=dp.get_precision('Product Unit of Measure'), required=True, readonly=True, states={'draft': [('readonly', False)]}),
+        'date_planned': fields.datetime('Scheduled Date', required=True, select=True, readonly=True, states={'draft': [('readonly', False)]}),
+        'taxes_id': fields.many2many('account.tax', 'purchase_order_taxe', 'ord_id', 'tax_id', 'Taxes', readonly=True, states={'draft': [('readonly', False)]}),
+        'product_uom': fields.many2one('product.uom', 'Product Unit of Measure', required=True, readonly=True, states={'draft': [('readonly', False)]}),
+        'product_id': fields.many2one('product.product', 'Product', domain=[('purchase_ok','=',True)], change_default=True, readonly=True, states={'draft': [('readonly', False)]}),
         'move_ids': fields.one2many('stock.move', 'purchase_line_id', 'Reservation', readonly=True, ondelete='set null'),
-        'price_unit': fields.float('Unit Price', required=True, digits_compute= dp.get_precision('Product Price')),
+        'price_unit': fields.float('Unit Price', required=True, digits_compute= dp.get_precision('Product Price'), readonly=True, states={'draft': [('readonly', False)]}),
         'price_subtotal': fields.function(_amount_line, string='Subtotal', digits=0),
-        'order_id': fields.many2one('purchase.order', 'Order Reference', select=True, required=True, ondelete='cascade'),
-        'account_analytic_id':fields.many2one('account.analytic.account', 'Analytic Account',),
+        'order_id': fields.many2one('purchase.order', 'Order Reference', select=True, required=True, ondelete='cascade', readonly=True, states={'draft': [('readonly', False)]}),
```

Comment:
I would suggest to add a domain [('state','=','draft')] to restrict users from changing PO when it's in other than "draft" state.